File: MyAnimeListJava/src/main/java/com/example/myanimelistjava/repositories/admin/AdminRepository.java
package com.example.myanimelistjava.repositories.admin;

import com.example.myanimelistjava.managers.DataBaseManager;
import com.example.myanimelistjava.models.Admin;
import com.example.myanimelistjava.models.User;
import com.example.myanimelistjava.repositories.users.IUserRepository;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.List;
import java.util.UUID;


public class AdminRepository implements IAdminRepository {

    private DataBaseManager db;

    @Override
    public Admin findById(Admin id) throws SQLException {
        String query = "select * from admins where id = ?";

/*
            ResultSet result = db.select(query, id);
            if (result.next()) {
                Admin admin = new Admin(

                );
*
                return admin;

        }*/
        return null;
    }

    @Override
    public List<Admin> findAll() throws SQLException {
        return null;
    }

    @Override
    public UUID update(UUID item) throws SQLException {
        return null;
    }

    @Override
    public UUID add(UUID item) throws SQLException {
        return null;
    }

    @Override
    public void delete(Admin id) throws SQLException {

    }
}

File: MyAnimeList/src/main/kotlin/com/example/myanimelist/repositories/AdminRepository.kt
package com.example.myanimelist.repositories

import com.example.myanimelist.managers.DataBaseManager
import com.example.myanimelist.models.*
import java.util.*

class AdminRepository(val db: DataBaseManager) : IAdminRepository {
    override fun findbyId(id: UUID): Admin? {
        var query = "select * from usuarios where id = ?"
        val result = db.select(query, id).get()
        if (result.first()) {
            val admin = Admin(
                UUID.fromString(result.getString("id")),
                result.getString("nombre"),
                result.getString("email"),
                result.getString("password"),
                result.getDate("date_alta"),
                result.getDate("date_nacimiento")
            )
            db.close()
            return admin
        }
        return null
    }

    override fun findAll(): List<Admin> {
        var query = "select * from usuarios"
        val result = db.select(query, id)
        while(result.next()){

        }
        if (result.first()) {
            val admin = Admin(
                UUID.fromString(result.getString("id")),
                result.getString("nombre"),
                result.getString("email"),
                result.getString("password"),
                result.getDate("date_alta"),
                result.getDate("date_nacimiento")
            )
            db.close()
            return admin
        }
        return null
    }

    override fun update(item: Admin?): Admin? {
        TODO("Not yet implemented")
    }

    override fun add(item: Admin?): Admin? {
        TODO("Not yet implemented")
    }


    override fun delete(id: UUID): Admin {
        TODO("Not yet implemented")
    }
}